osgp/platform/osgp-secret-management/src/main/java/org/opensmartgridplatform/secretmanagement/application/repository/DbEncryptedSecretRepository.java

```diff
@@ -30,6 +30,18 @@
       @Param("secretType") SecretType secretType,
       @Param("secretStatus") SecretStatus secretStatus);
 
+  @Query(
+      value =
+          "SELECT es FROM DbEncryptedSecret es "
+              + "WHERE es.deviceIdentification = :deviceIdentification AND es.secretType = :secretType "
+              + "AND es.secretStatus = :secretStatus AND es.encodedSecret = :encodedSecret "
+              + "ORDER BY es.creationTime DESC, es.id DESC")
+  List<DbEncryptedSecret> findSecretsIncludingEncodedSecret(
```

Comment:
All other attributes (so excluding the EncodedSecret) are part of the public/logical key. This query in fact checks whether the secret exists with the actual value of the key. It might be more clear to name the method `findSecretsWithSpecificEncodedSecret`